.docs/extensions/package_docs.py

```diff
@@ -58,24 +89,31 @@ def _package_docs(app: sphinx.application.Sphinx) -> None:
 
 {package}
 {underline}
+""".strip()
+AUTOMODULE_TEMPLATE = """
 
 .. automodule:: {package}
-""".strip()
+""".rstrip()
 
 
-def _main(docs_dir: pathlib.Path, package: str) -> None:
-    subdir, _, package_dir_name = package.rpartition('/')
-    subdir = subdir or '.'
-    generated_dir = docs_dir / 'reference' / 'charmlibs' / subdir
-    generated_dir.mkdir(parents=True, exist_ok=True)
+def _main(docs_dir: pathlib.Path, package: str | None) -> None:
+    """Write automodule file for package and placeholders rst files for all other packages."""
     root = docs_dir.parent
-    assert (root / subdir / package_dir_name).is_dir()
-    prefix = 'charmlibs.' if subdir == '.' else f'charmlibs.{subdir}.'
-    module = package_dir_name.replace('-', '_')
-    content = AUTOMODULE_TEMPLATE.format(
-        prefix=prefix, package=module, underline='=' * len(package)
-    )
-    _write_if_needed(path=generated_dir / f'{package}.rst', content=content)
+    ref_dir = docs_dir / 'reference'
+    (ref_dir / 'charmlibs' / 'interfaces').mkdir(parents=True, exist_ok=True)
+    for subdir, p in (
+        *(('', p.name) for p in root.glob(r'[a-z]*') if p.is_dir() and p.name != 'interfaces'),
+        *(('interfaces', p.name) for p in (root / 'interfaces').glob(r'[a-z]*') if p.is_dir()),
+    ):
```

Comment:
If you don't already know what is going on, I think this part is difficult to approach. How about adding the same comment as in docs.just? I even wonder about extracting this part into a single-function file that can be imported in both places, but maybe that's overkill. A comment ought to be enough I think

```suggestion
    # Any directory (or subdirectory of interfaces) starting with a-z is assumed to be a package.
    for subdir, p in (
        *(('', p.name) for p in root.glob(r'[a-z]*') if p.is_dir() and p.name != 'interfaces'),
        *(('interfaces', p.name) for p in (root / 'interfaces').glob(r'[a-z]*') if p.is_dir()),
    ):
```